src/CMakeLists.txt

```diff
@@ -131,6 +131,36 @@ ign_build_tests(TYPE UNIT
     ignition-gazebo${PROJECT_VERSION_MAJOR}
 )
 
+if(TARGET UNIT_ign_TEST)
+
+  # Running `ign gazebo` on macOS has problems when run with /usr/bin/ruby
+  # due to System Integrity Protection (SIP). Try to find ruby from
+  # homebrew as a workaround.
+  if (APPLE)
+    find_program(BREW_RUBY ruby HINTS /usr/local/opt/ruby/bin)
+  endif()
+
+  add_dependencies(UNIT_ign_TEST
+    ${ign_lib_target}
+    TestModelSystem
+    TestSensorSystem
+    TestWorldSystem
+  )
+
+  target_compile_definitions(UNIT_ign_TEST PRIVATE
+      "BREW_RUBY=\"${BREW_RUBY} \"")
+
+  target_compile_definitions(UNIT_ign_TEST PRIVATE
+      "IGN_PATH=\"${IGNITION-TOOLS_BINARY_DIRS}\"")
+
+  set(_env_vars)
+  list(APPEND _env_vars "IGN_CONFIG_PATH=${CMAKE_BINARY_DIR}/test/conf")
+  list(APPEND _env_vars "IGN_GAZEBO_SYSTEM_PLUGIN_PATH=$<TARGET_FILE_DIR:TestModelSystem>")
+
+  set_tests_properties(UNIT_ign_TEST PROPERTIES
+    ENVIRONMENT "${_env_vars}")
```

Comment:
Interesting, I don't think we've been setting environment variables for tests from CMake.

Reply:
I learned it from Dirk; it's not universal, but we have done it in a few places:

* https://github.com/ignitionrobotics/ign-cmake/blob/ignition-cmake2_2.6.0/examples/CMakeLists.txt#L180-L186
* https://github.com/osrf/sdformat/blob/sdformat10_10.0.0/cmake/SDFUtils.cmake#L192-L197
* https://github.com/osrf/gazebo/blob/gazebo11_11.3.0/cmake/GazeboTestUtils.cmake#L69-L77

I think it's a bit cleaner to set them from cmake. One drawback is that executing the binaries directly (like `./UNIT_ign_TEST`) may not work, but it's easy enough to use `ctest -R UNIT_ign_TEST -VV` once you know to do so

Reply:
>  One drawback is that executing the binaries directly (like ./UNIT_ign_TEST) may not work

Oh no, that sounds like a regression. We have that workflow documented in a few places, it would be unfortunate not to support it anymore.

Reply:
As I mentioned during the meeting, I don't see the advantage of moving the environment variables to CMake. 

It prevents a common workflow, and it hides the environment configuration from anyone who's debugging the tests or looking into them to understand how the code works. Especially for variables like `IGN_GAZEBO_SYSTEM_PLUGIN_PATH`, which are set by the user at runtime, I think it's valuable to explicitly set them during the tests.

But I'm not going to block this PR on this, I'll leave it up to you.

Reply:
I pointed to 3 examples in our code where it's already being used, so this isn't a completely new thing, but you're right that the docs need to be updated to handle this case. I'll merge this then open a pull request to update the docs